tidy3d/plugins/invdes/postprocess.py

```diff
@@ -0,0 +1,131 @@
+
+    monitor_name: str = pd.Field(
+        ...,
+        title="Monitor Name",
+        description="Name of the ``ModeMonitor`` corresponding to the ``ModeData`` "
+        "that we want to compute power for.",
+    )
+
+    direction: td.components.types.Direction = pd.Field(
+        None,
+        title="Direction",
+        description="If specified, selects a specific direction ``'-'`` or ``'+'`` from the "
+        "``ModeData`` to include in power. Otherwise, sums over all directions.",
+    )
+
+    mode_index: pd.NonNegativeInt = pd.Field(
+        None,
+        title="Mode Index",
+        description="If specified, selects a specific mode index from the ``ModeData`` "
+        "to include in power. Otherwise, sums over all mode indices.",
+    )
+
+    f: float = pd.Field(
+        None,
+        title="Frequency",
+        description="If specified, selects a specific frequency from the ``ModeData`` "
+        "to include in power. Otherwise, sums over all frequencies.",
+    )
+
+    weight: float = pd.Field(
+        1.0,
+        title="Weight",
+        description="Weight specifying the contribution of this power to the objective function. ",
+    )
+
+    @property
+    def sel_kwargs(self) -> dict[str, typing.Any]:
+        """Selection kwargs corresponding to the fields."""
+        sel_kwargs_all = dict(direction=self.direction, mode_index=self.mode_index, f=self.f)
+        return {key: sel for key, sel in sel_kwargs_all.items() if sel is not None}
+
+    def evaluate(self, sim_data: td.SimulationData) -> float:
+        """Evaluate this instance when passed a simulation dataset."""
+
+        mnt_data = sim_data[self.monitor_name]
+
+        if not isinstance(mnt_data, td.ModeData):
+            raise ValueError(
+                "'ModePower' only works with 'ModeData' corresponding to 'ModeMonitor'. "
+                f"Monitor name of '{self.monitor_name}' returned data of type {type(mnt_data)}."
+            )
+
+        amps = mnt_data.amps
+        amp = amps.sel(**self.sel_kwargs)
+        powers = abs(amp.values) ** 2
+        power = anp.sum(powers)
+        return self.weight * power
+
+
+ElementaryPostprocessOperationType = typing.Union[ModePower]
+
+
+class CombinedPostprocessOperation(AbstractPostprocessOperation, abc.ABC):
+    """A postprocess operation that combines elementary operations."""
+
+    operations: tuple[ElementaryPostprocessOperationType, ...] = pd.Field(
+        (),
+        title="Postprocessing Operations",
+        description="Set of objects specifying the operations combined in this operation.",
+    )
+
+
+class Sum(CombinedPostprocessOperation):
+    """Sum of the evaluate outputs of elementary operation objects."""
+
+    def evaluate(self, sim_data: td.SimulationData) -> float:
+        """sum the evaluation of each operation."""
+
+        value = 0.0
+        for operation in self.operations:
+            value = value + operation.evaluate(sim_data)
+
+        return value
```

Comment:
Maybe `return sum(op.evaluate(sim_data) for op in self.operations)`?

Reply:
That works. Originally I applied the weights here, which is why I had it written out like this.